### PR TITLE
8322817: RISC-V: Eliminate -Wparentheses warnings in riscv code

### PR DESCRIPTION
**Pull request by kimbarrett**
Please review this change to eliminate some -Wparentheses warnings. This
involved simply adding a few parentheses to make some implicit operator
precedence explicit.

Testing: Local (linux-x64) cross-build for linux-riscv with this change plus
-Wparentheses enabled and other changes to allow that to work.

Requesting someone from the riscv porters to properly test this.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8322817](https://bugs.openjdk.org/browse/JDK-8322817): RISC-V: Eliminate -Wparentheses warnings in riscv code (**Enhancement** - P4)

### Reviewers
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**) ⚠️ Review applies to [ef64e2e1](https://git.openjdk.org/jdk/pull/17216/files/ef64e2e1470eae32688ca9aa025550fe43507274)
 * [Ludovic Henry](https://openjdk.org/census#luhenry) (@luhenry - Committer) ⚠️ Review applies to [ef64e2e1](https://git.openjdk.org/jdk/pull/17216/files/ef64e2e1470eae32688ca9aa025550fe43507274)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17216/head:pull/17216` \
`$ git checkout pull/17216`

Update a local copy of the PR: \
`$ git checkout pull/17216` \
`$ git pull https://git.openjdk.org/jdk.git pull/17216/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17216`

View PR using the GUI difftool: \
`$ git pr show -t 17216`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17216.diff">https://git.openjdk.org/jdk/pull/17216.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17216#issuecomment-1873710043)

### CODE REVIEW

**Comment by zifeihan** [comment]
Hi, I am performing a tier1 test with fastdebug build on linux-riscv64.

**Review by RealFYang — APPROVED** [review]
LGTM. Thanks.

**Comment by zifeihan** [comment]
Hi, tier1 passed on qemu 8.1.0 with UseRVV (fastdebug)

**Review by luhenry — APPROVED** [review]

File: src/hotspot/cpu/riscv/frame_riscv.inline.hpp [hunk 1]
```diff
@@ -186,7 +186,7 @@ inline bool frame::equal(frame other) const {
               unextended_sp() == other.unextended_sp() &&
               fp() == other.fp() &&
               pc() == other.pc();
-  assert(!ret || ret && cb() == other.cb() && _deopt_state == other._deopt_state, "inconsistent construction");
+  assert(!ret || (cb() == other.cb() && _deopt_state == other._deopt_state), "inconsistent construction");
```
Comment (kimbarrett) [inline src/hotspot/cpu/riscv/frame_riscv.inline.hpp:189]: Per comments in reviews for similar changes for other ports, I've removed the unnecessary "ret &&".
Builds fine, but should probably be (re)tested by a port maintainer.

**Comment by zifeihan** [comment]
Hi, the latest chang still test good with fastdebug build.

**Comment by kimbarrett** [comment]
Thanks for reviews, @luhenry and @RealFYang .  Thanks for testing, @zifeihan .

**Pull request closed without merging**